chpl/chpl-service/src/main/java/gov/healthit/chpl/caching/CacheInitializorImpl.java

```diff
@@ -0,0 +1,107 @@
+package gov.healthit.chpl.caching;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import gov.healthit.chpl.dao.CertificationStatusDAO;
+import gov.healthit.chpl.dao.EntityRetrievalException;
+import gov.healthit.chpl.dao.PendingCertifiedProductDAO;
+import gov.healthit.chpl.dao.impl.CertifiedProductSearchResultDAOImpl;
+import gov.healthit.chpl.domain.SearchRequest;
+import gov.healthit.chpl.domain.SurveillanceSearchOptions;
+import gov.healthit.chpl.dto.CertificationStatusDTO;
+import gov.healthit.chpl.manager.impl.CertificationIdManagerImpl;
+import gov.healthit.chpl.manager.impl.SearchMenuManagerImpl;
+
+@Component
+public class CacheInitializorImpl {
```

Comment:
Asynchronous execution is weird and requires being called from another class in order to run on another thread, so I can't move CacheInitializorImpl into CacheInitializor. I can definitely rename to something more specific, though.